test/runtime_tests.cc

```diff
@@ -992,5 +995,30 @@ namespace detail {
 		});
 	}
 
+	void dry_run_with_nodes(const size_t nodes) {
+		const std::string dryrun_envvar_name = "CELERITY_DRYRUN_NODES";
+		const auto ste = test_utils::set_test_env(dryrun_envvar_name, std::to_string(nodes));
+
+		distr_queue q;
+
+		auto& rt = runtime::get_instance();
+		auto& tm = rt.get_task_manager();
+		tm.set_horizon_step(2);
+
+		CHECK(rt.is_dry_run());
+
+		q.submit([=](handler& cgh) { cgh.host_task(range<1>{nodes * 2}, [](partition<1>) {}); });
+		q.slow_full_sync();
+
+		// (intial epoch + task + sync epoch) per node.
+		CHECK(runtime_testspy::get_command_count(rt) == 3 * nodes);
+		test_utils::maybe_print_graph(tm);
+	}
+
+	TEST_CASE_METHOD(test_utils::runtime_fixture, "Dry run generates commands for an arbitrary number of simulated worker nodes", "[dryrun]") {
+		const size_t nodes = GENERATE(values({4, 8, 16}));
+		SECTION("with {} nodes", std::to_string(nodes).c_str()) { dry_run_with_nodes(nodes); }
```

Comment:
```suggestion
		SECTION("with " << nodes << " nodes", dry_run_with_nodes(nodes); }
```

Catch2 does not support fmt-style formatting.